Core/DTO/CategoryDTO.cs

```diff
@@ -0,0 +1,52 @@
+﻿using LearnNet_CatalogService.Data.Entities;
+
+namespace LearnNet_CatalogService.Core.DTO
+{
+    public class CategoryDTO
+    {
+        public int Id { get; set; }
+        public required string Name { get; set; }
+        public Uri? ImageUrl { get; set; }
+
+        public CategoryDTO? ParentCategory { get; set; }
+        public int ParentCategoryId { get; set; }
+
+        public static Category? MapTo(CategoryDTO? dto)
+        {
+            if (dto == null)
+            {
+                return null;
+            }
+
+            var entity = new Category
+            {
```

Comment:
null checks?

Reply:
please add null checks to all mappings

Reply:
not sure what exactly additional do you mean

Reply:
Nevermind, my mistake